app/stac_api/pgtriggers.py

```diff
@@ -461,3 +470,54 @@ def generates_asset_upload_triggers():
             func=etag_func
         ),
     ]
+
+
+class SummaryFields(Enum):
+    # pylint: disable=invalid-name
+    GSD = 'summaries_eo_gsd', 'gsdcount'
+    LANGUAGE = 'summaries_geoadmin_lang', 'geoadminlangcount'
+    VARIANT = 'summaries_geoadmin_variant', 'geoadminvariantcount'
+    PROJ_EPSG = 'summaries_proj_epsg', 'projepsgcount'
+
+
+def generate_summary_count_triggers(summary_field, count_table):
+
+    class UpdateCollectionSummaryTrigger(pgtrigger.Trigger):
+        when = pgtrigger.After
+        declare = [
+            ('count_instance', f'stac_api_{count_table}%ROWTYPE'),
+            ('collection_summaries', 'RECORD'),
+        ]
+        func = f'''
+        count_instance = COALESCE(NEW, OLD);
+
+        -- Compute collection summaries
+        SELECT
+            collection_id,
+            array_agg(value) AS values
+        INTO collection_summaries
+        FROM stac_api_{count_table}
+        WHERE count > 0 AND value IS NOT NULL AND collection_id = count_instance.collection_id
+        GROUP BY collection_id;
+
+        -- Update related collection (auto variables + summaries)
+        UPDATE stac_api_collection
+        SET {summary_field} = COALESCE(collection_summaries.values, '{{}}')
+        WHERE id = count_instance.collection_id;
+
+        RAISE INFO 'collection.id=% summaries updated',
+            count_instance.collection_id;
+        RETURN count_instance;
+        '''
+
+    return [
+        UpdateCollectionSummaryTrigger(
+            name=f'update_collection_{count_table}_trigger',
+            operation=pgtrigger.Update,
+            condition=pgtrigger.Condition('OLD.count < 2 OR NEW.count < 2')
```

Comment:
not sure if i understand the purpose of this condition here but the trigger will only be fired if the old or the new count is smaller than 2? why is that necessary?

Reply:
The idea is that we don't need to take the aggregate list again if the count changed from like 3 to 4, as there is no new entry. we are only really interested in the case where the count changes from 0 to 1 and vice versa.
I slightly changed the Condition and added a comment to try and make it clearer.